### ISSUE TITLE
support atomic swaps as defined by github.com/decred/atomicswap

### ISSUE THREAD

**Issue by GlenDC** [threefoldfoundation/tfchain#18]
Atomic swaps allow the exchange of tokens between 2 different ledgers, using no outside validators, and thus are concerned only to the two parties involved. An exchange of information happens over a side-channel. The side-channel can be as simple as a chat service, but that is a detail not solved by decred/atomicswap or us. The transactions of both coin X and coin Y are both tracked on chain on their relative blockchain.

As decred/atomicswap is already supported (and maintained) for bitcoin (and friends), we simply need to support this in Rivine as well, and we are set to go. Thus 60% of the work is already done (as the design is OK, and the other ledger (e.g. bitcoin) already supports it).

**Comment by GlenDC** [comment]
Before we can implement the feature for this issue, we are still doing all the necessary preparation work on Rivine side. See https://github.com/rivine/rivine/issues/142.
